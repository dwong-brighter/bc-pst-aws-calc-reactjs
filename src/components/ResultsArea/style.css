.ResultsArea-container {
  align-content: left;
  display: block;
  background-color: lightblue;
  border-radius: 10px;
  margin: 20px 10px;
  padding: 10px 5px;
}
