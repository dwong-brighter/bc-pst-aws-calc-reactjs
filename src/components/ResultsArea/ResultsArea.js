import React from 'react';
import CalcEngine from '../../helpers/CalcEngine';

/*
Can't do the following at the moment due to
https://github.com/webpack-contrib/sass-loader/issues/206
and
https://github.com/danedavid/react-redux-boilerplate/issues/1
import style from './style.css';

And then in the code you would have
<theElement className={style.theStyle}></theElement>
*/
import './style.css';

export default function ResultsArea (props) {
  const {exchangeRate, usdTotalAmount, usdGstAmount} = props.data;
  const usdNetAmount = CalcEngine.calculateUsdNetCost(usdTotalAmount, usdGstAmount);
  const usdNetAmountGstAndPstTaxable = CalcEngine.calculateUsdNetCostSubjectToGstAndPst(usdGstAmount);
  const usdNetAmountOnlyPstTaxable = CalcEngine.calculateUsdNetCostSubjectToPstOnly(usdTotalAmount, usdGstAmount);
  const usdPstAmount = CalcEngine.calculateUsdPst(usdTotalAmount, usdGstAmount);

  return (
    <div className='ResultsArea-container'>
      <p>USD to CAD Exchange Rate is <strong>{exchangeRate.toFixed(14)}</strong></p>
      <p>USD Total of <strong>${usdTotalAmount.toFixed(2)}</strong> is <strong>${CalcEngine.convertUsdToCad(usdTotalAmount, exchangeRate).toFixed(2)}</strong> CAD</p>
      <p>USD Net Price Subject to GST and PST of <strong>${usdNetAmountGstAndPstTaxable.toFixed(2)}</strong> is <strong>${CalcEngine.convertUsdToCad(usdNetAmountGstAndPstTaxable, exchangeRate).toFixed(2)}</strong> CAD</p>
      <p>USD Net Price Subject to PST only of <strong>${usdNetAmountOnlyPstTaxable.toFixed(2)}</strong> is <strong>${CalcEngine.convertUsdToCad(usdNetAmountOnlyPstTaxable, exchangeRate).toFixed(2)}</strong> CAD</p>
      <p>USD Net Cost of (all subject to PST) <strong>${usdNetAmount.toFixed(2)}</strong> is <strong>${CalcEngine.convertUsdToCad(usdNetAmount, exchangeRate).toFixed(2)}</strong> CAD</p>
      <p>USD GST of <strong>${usdGstAmount.toFixed(2)}</strong> is <strong>${CalcEngine.convertUsdToCad(usdGstAmount, exchangeRate).toFixed(2)}</strong> CAD</p>
      <p>USD PST to Self-Assess of <strong>${usdPstAmount.toFixed(2)}</strong> is <strong>${CalcEngine.convertUsdToCad(usdPstAmount, exchangeRate).toFixed(2)}</strong> CAD</p>
    </div>
  );
}
