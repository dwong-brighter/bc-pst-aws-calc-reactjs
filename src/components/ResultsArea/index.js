import ResultsArea from './ResultsArea.js'

export default ResultsArea
