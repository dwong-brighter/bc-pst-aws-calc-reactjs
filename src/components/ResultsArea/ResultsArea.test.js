import React from 'react';
import { shallow } from 'enzyme';

import ResultsArea from './ResultsArea.js';

it('needs props to render', () => {
  const TEST_DATA = {
    exchangeRate: 1.2,
    usdTotalAmount: 155,
    usdGstAmount: 5
  };

  shallow(<ResultsArea data={TEST_DATA} />);
});
