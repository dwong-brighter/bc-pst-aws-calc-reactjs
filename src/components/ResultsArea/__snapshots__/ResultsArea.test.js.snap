// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ResultsArea /> renders without crashing 1`] = `
<Styled(MuiBox)
  className="makeStyles-container-1"
>
  <p>
    USD to CAD Exchange Rate is 
    <strong>
      4.20000000000000
    </strong>
  </p>
  <p>
    USD Total of 
    <strong>
      $
      42.00
    </strong>
     is 
    <strong>
      $
      176.40
    </strong>
     CAD
  </p>
  <p>
    USD Net Price Subject to GST and PST of 
    <strong>
      $
      840.00
    </strong>
     is 
    <strong>
      $
      3528.00
    </strong>
     CAD
  </p>
  <p>
    USD Net Price Subject to PST only of 
    <strong>
      $
      -840.00
    </strong>
     is 
    <strong>
      $
      -3528.00
    </strong>
     CAD
  </p>
  <p>
    USD Net Cost of (all subject to PST) 
    <strong>
      $
      0.00
    </strong>
     is 
    <strong>
      $
      0.00
    </strong>
     CAD
  </p>
  <p>
    USD GST of 
    <strong>
      $
      42.00
    </strong>
     is 
    <strong>
      $
      176.40
    </strong>
     CAD
  </p>
  <p>
    USD PST to Self-Assess of 
    <strong>
      $
      0.00
    </strong>
     is 
    <strong>
      $
      0.00
    </strong>
     CAD
  </p>
  <p>
    <em>
      CAD Total Rounding is 
      <strong>
        $
        0.42
      </strong>
       CAD
    </em>
  </p>
  <p>
    <em>
      CAD GST Rounding is 
      <strong>
        $
        0.42
      </strong>
       CAD
    </em>
  </p>
</Styled(MuiBox)>
`;
